### PR TITLE
Finished encrypted file ex

### PR DESCRIPTION
**Pull request by michall1900**
Hey itayday and talbarfm,
Please check the exercise,
Thanks in advance .

### CODE REVIEW

File: 5.3/Encrypted logo.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+The binary file is encrypted and there are some secret lines.
+Secret lines are the once that end with !, their len is >=5 and
+they include just lower letters.
+The program find and print secrete lines.
+"""
+import typing
+
+
+def is_encrypted_line(line: str) -> bool:
+    """ Return if the line's len is bigger than / equal to 5 and ends with !.
```
Comment (talbarfm) [inline 5.3/Encrypted logo.py]: newline after """

File: 5.3/Encrypted logo.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+""" The program open logo file in binary style.
```
Comment (talbarfm) [inline 5.3/Encrypted logo.py]: opens*

File: 5.3/Encrypted logo.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+The binary file is encrypted and there are some secret lines.
+Secret lines are the once that end with !, their len is >=5 and
+they include just lower letters.
+The program find and print secrete lines.
```
Comment (talbarfm) [inline 5.3/Encrypted logo.py]: This documentation is good, but its redundant. document the relevant parts not the head of the page

File: 5.3/Encrypted logo.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+"""
+import typing
+
+
+def is_encrypted_line(line: str) -> bool:
+    """ Return if the line's len is bigger than / equal to 5 and ends with !.
+    :param line: string with only lower letters beside the last one.
+    :return: true if it is secret line or not.
+    """
+    if len(line) < 5 or not line.endswith('!'):
```
Comment (talbarfm) [inline 5.3/Encrypted logo.py]: you can just do this - 
return not (len(line) < 5 or not line.endswith('!'))

File: 5.3/Encrypted logo.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+
+def find_encrypted_lines_generator(encrypted_binary_file: typing.BinaryIO) -> str:
+    """generator that return all of the secrete lines.
+    :param encrypted_binary_file to decrypt it with format described before
+    :return: secrete lines
+    """
+    curr_word = ""
+    byte = (encrypted_binary_file.read(1))
+    while byte:
```
Comment (talbarfm) [inline 5.3/Encrypted logo.py]: If you want, I think solving this question will be easier to you if you use the regex module

File: 5.3/Encrypted logo.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    return True
+
+
+def find_encrypted_lines_generator(encrypted_binary_file: typing.BinaryIO) -> str:
+    """generator that return all of the secrete lines.
+    :param encrypted_binary_file to decrypt it with format described before
+    :return: secrete lines
+    """
+    curr_word = ""
+    byte = (encrypted_binary_file.read(1))
```
Comment (talbarfm) [inline 5.3/Encrypted logo.py]: give a better name then byte

File: 5.3/Encrypted logo.py [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    while byte:
+        curr_char = byte.decode(errors="ignore")
```
Comment (talbarfm) [inline 5.3/Encrypted logo.py]: dont use abbreviations - current_word, current_char

**Review by talbarfm — COMMENTED** [review]
Good job!!

**Comment by talbarfm** [comment]
Give more detailed commit messages

**Review by lovodkin93 — APPROVED** [review]
the code looks very good. Well done!

**Pull request merged**